client/securedrop_client/sdk/__init__.py

```diff
@@ -328,9 +334,12 @@ def _send_json_request(
         if self.development_mode:
             env["SD_PROXY_ORIGIN"] = self.server
 
+        if not progress:
+            progress = ProgressProxy(None)
```

Comment:
This is a dummy progressproxy that doesn't send update signals anywhere, in order to satisfy the signature of `_streaming_download` ?

Since there are other places where you have `ProgressProxy | None`, and this could be a bit confusing, I wonder about at minimum adding a comment to explain, but potentially changing the signature so that the sdk is always guaranteed to be passed a progressproxy (and therefore punting the decision strictly to the ui), or doing the opposite and accepting Proxy | None all the way through the sdk. No strong feelings just thinking aloud.

Reply:
Yeah. It's definitely inconsistent, my thinking was that _streaming_download is already such a complicated and nested function that let's not add anymore complexity to it...but I was really just saving:

```py
if progress:
    progress.set_value(bytes_written)
```

which isn't really that bad, vs having the extra load of inner being None.